lib/collection/src/collection/shard_transfer.rs

```diff
@@ -418,7 +420,15 @@ impl Collection {
             }
 
             if replica_set.is_dummy().await {
+                // Check if shard was dirty before init_empty_local_shard
+                let was_dirty = replica_set.is_dirty().await;
+                // TODO: If dirty, still try to load the shard and init empty shard only if it's not recoverable?
                 replica_set.init_empty_local_shard().await?;
```

Comment:
But! Inside `init_empty_local_shard` we must do `LocalShard::clear()` to actually remove the corrupt files (if there are any). There's a TODO in there which is probably the right place to put it.

Reply:
Done inside `init_empty_local_shard()` since it was only used here.